impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -185,10 +185,20 @@ private CredentialValidationResult checkThroughSearch(UsernamePasswordCredential
                         null, // caller unique id
                         groups);
 
-            } catch (IllegalStateException e) {
-                return NOT_VALIDATED_RESULT;
+            } catch (Exception e) {
```

Comment:
I think the catch clause can't go here -- there are two attempts to create an LdapContext inside the try, and the meaning of an exception will be different depending on which one cause the exception. In the first case, we're binding using credentials supplied by configuration. So an authentication problem there means a bad configuration. In the second case, we looked up the user and we're trying to bind as that user in order to validate the password. An authentication problem there means a bad password. We'll need to check errors much closer to the source to make sure we respond appropriately.